pkg/did/creator/creator.go

```diff
@@ -105,3 +124,54 @@ func (d *Creator) createDIDKeyDoc(createDIDOpts *api.CreateDIDOpts) (*did.DocRes
 
 	return didkeycreator.NewCreator().Create(key, verificationType)
 }
+
+func (d *Creator) createDIDIonLongFormDoc(createDIDOpts *api.CreateDIDOpts) (*did.DocResolution, error) {
+	var (
+		key []byte
+		vm  *did.VerificationMethod
+		err error
+	)
+
+	// TODO: refactor so more code is shared between handlers for different did methods
+	if d.keyReader == nil { //nolint: nestif
+		// Generate a key and set the verification type in behalf of the caller.
+		var keyID string
+
+		keyID, key, err = d.keyWriter.Create(arieskms.ED25519Type)
+		if err != nil {
+			return nil, err
+		}
+
+		vm = &did.VerificationMethod{ID: "#" + keyID, Value: key, Type: Ed25519VerificationKey2018}
+	} else { // Use the caller's chosen key and verification type.
+		if createDIDOpts.VerificationType == "" {
+			return nil, errors.New("no verification type specified")
+		}
+
+		key, err = d.keyReader.ExportPubKey(createDIDOpts.KeyID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get key handle: %w", err)
```

Comment:
I guess now it's actually just the public key, not a "key handle" anymore (again, in the did:key implementation I did I have the same message - I think since then we've reworked that keyReader interface a bit)